options.go

```diff
@@ -18,6 +18,18 @@ func WithSourceFiles(files ...string) Option {
 	}
 }
 
+// WithSourceLoaders sets the source loaders to use to load the files.
+// It can be used multiple times to load from multiple loaders.
+// The first loader in the list takes the lowest precedence, and the last loader takes the highest precedence.
+// They take a higher precedence than SourceFiles.
+// E.g. if the same key is defined in both files, the value in the last loader will be used.
+// Base -> Override 1 -> Override 2 -> Override 3
+func WithSourceLoaders(loader ...Loader) Option {
+	return func(s *Service) {
+		s.SourceLoaders = append(s.SourceLoaders, loader...)
```

Comment:
I think it would be nice if we kept the env loader as the highest level of overriding. I can see this functionality being used to pull from cloud key vaults and I would say that this should be given the same precedence as a standard config file. It might be nice in a future PR to allow users to specify the order of precedence?

Reply:
I agree - resolved